### PR TITLE
Reflector fixes and improvements

### PR DESCRIPTION
**Pull request by ACCount12**
This PR contains multiple reflector fixes and QoL improvements.
default
* Reflector sprites are changed. They have a base plate now, copied from solars. Improvements are welcome.
* Multiple lines of text added or changed. Improvements are welcome.
* Reflector rotation window now displays the current rotation as default value.
* You can lock reflector's rotation by using a screwdriver. Use the screwdriver again to unlock it.
* Reflector examine now shows the current rotation and rotation lock status.
* You can now use a welder to repair damaged reflectors. This is really helpful, given how often they get damaged by lasers if you tamper with reflector setup.
* Fixes a bug when reflector's sprite was pointing in a direction different from its actual reflection direction.
* Fixes a bug with deflectors being deconstructable when welded down.
* Fixes a `NORM_ROT` bug where `0` was normalized as `360`.

:cl: ACCount
imageadd: New sprites for reflectors. They finally look like something. 
add: You can lock reflector's rotation by using a screwdriver. Use the screwdriver again to unlock it.
add: Reflector examine now shows the current angle and rotation lock status.
add: You can now use a welder to repair damaged reflectors.
fix: Multiple reflector bugs are fixed.
/:cl:

### CODE REVIEW

**Comment by WJohn** [comment]
These look way better. Nice work!

**Comment by ACCount12** [comment]
I'm pretty good at coderspriting. Still wouldn't mind some help from a real spriter. 

Regular and two-side reflectors are fine, but "box" resprite looks ugly when turned at angles that are not % 45 and I have no idea how to fix it.
default

**Comment by ShizCalev** [comment]
Seeing as you've removed all the directional sprites that were added in #31863, can you confirm that you can see what way reflectors are pointing in DM now?

**Comment by ACCount12** [comment]
There wasn't. Now there is.

File: code/game/objects/structures/reflector.dm [hunk 1]
```diff
@@ -51,30 +77,53 @@
 /obj/structure/reflector/attackby(obj/item/W, mob/user, params)
 	if(admin)
 		return
+
+	if(istype(W, /obj/item/screwdriver))
+		can_rotate = !can_rotate
+		to_chat(user, "<span class='notice'>You [can_rotate ? "unlock" : "lock"] [src]'s rotation.</span>")
+		playsound(loc, W.usesound, 50, 1)
```
Comment (Cyberboss) [inline code/game/objects/structures/reflector.dm]: playsound(src, ...

File: code/game/objects/structures/reflector.dm [hunk 2]
```diff
@@ -51,30 +77,53 @@
...
 	if(istype(W, /obj/item/wrench))
 		if(anchored)
-			to_chat(user, "Unweld [src] first!")
+			to_chat(user, "<span class='warning'>Unweld [src] from the floor first!</span>")
+			return
+		user.visible_message("[user] starts to dismantle [src].", "<span class='notice'>You start to dismantle [src]...</span>")
 		if(do_after(user, 80*W.toolspeed, target = src))
-			playsound(src.loc, W.usesound, 50, 1)
-			to_chat(user, "You dismantle [src].")
+			playsound(loc, W.usesound, 50, 1)
```
Comment (Cyberboss) [inline code/game/objects/structures/reflector.dm]: ditto

File: code/game/objects/structures/reflector.dm [hunk 3]
```diff
@@ -51,30 +77,53 @@
...
 	else if(istype(W, /obj/item/weldingtool))
 		var/obj/item/weldingtool/WT = W
-		if(!anchored)
+
+		if(obj_integrity < max_integrity)
+			if(WT.remove_fuel(0,user))
+				user.visible_message("[user] starts to repair [src].",
+									"<span class='notice'>You begin repairing [src]...</span>",
+									"<span class='italics'>You hear welding.</span>")
+				playsound(loc, W.usesound, 40, 1)
```
Comment (Cyberboss) [inline code/game/objects/structures/reflector.dm]: ditto

File: code/game/objects/structures/reflector.dm [hunk 4]
```diff
@@ -51,30 +77,53 @@
...
+				if(do_after(user,40*WT.toolspeed, target = src))
+					obj_integrity = max_integrity
+					user.visible_message("[user.name] has repaired [src].", \
```
Comment (Cyberboss) [inline code/game/objects/structures/reflector.dm]: just [user]

File: code/game/objects/structures/reflector.dm [hunk 5]
```diff
@@ -1,13 +1,16 @@
...
+	name = "reflector base"
+	icon = 'icons/obj/structures.dmi'
+	icon_state = "reflector_map"
+	desc = "A base for reflector assemblies."
 	anchored = FALSE
-	density = TRUE
+	density = FALSE
 	layer = BELOW_OBJ_LAYER
+	var/deflector_icon_state = null
+	var/image/deflector_overlay = null
```
Comment (Cyberboss) [inline code/game/objects/structures/reflector.dm]: remoce the `= null`s

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by ACCount12** [comment]
All fixed. And please, set the tags before merging.

File: code/game/objects/structures/reflector.dm [hunk 6]
```diff
@@ -86,50 +135,44 @@
 		if(finished)
 			return
 		var/obj/item/stack/sheet/S = W
-		if(istype(W, /obj/item/stack/sheet/glass))
+		if(istype(S, /obj/item/stack/sheet/glass))
 			if(S.use(5))
-				new /obj/structure/reflector/single (loc)
-				qdel (src)
+				new /obj/structure/reflector/single(loc)
```
Comment (Cyberboss) [inline code/game/objects/structures/reflector.dm]: loc -> drop_location()

File: code/game/objects/structures/reflector.dm [hunk 7]
```diff
@@ -86,50 +135,44 @@
...
+				qdel(src)
 			else
 				to_chat(user, "<span class='warning'>You need five sheets of glass to create a reflector!</span>")
 				return
-		if(istype(W, /obj/item/stack/sheet/rglass))
+		if(istype(S, /obj/item/stack/sheet/rglass))
 			if(S.use(10))
-				new /obj/structure/reflector/double (loc)
+				new /obj/structure/reflector/double(loc)
```
Comment (Cyberboss) [inline code/game/objects/structures/reflector.dm]: loc -> drop_location()

File: code/game/objects/structures/reflector.dm [hunk 8]
```diff
@@ -86,50 +135,44 @@
...
 				qdel(src)
 			else
 				to_chat(user, "<span class='warning'>You need ten sheets of reinforced glass to create a double reflector!</span>")
 				return
-		if(istype(W, /obj/item/stack/sheet/mineral/diamond))
+		if(istype(S, /obj/item/stack/sheet/mineral/diamond))
 			if(S.use(1))
-				new /obj/structure/reflector/box (loc)
+				new /obj/structure/reflector/box(loc)
```
Comment (Cyberboss) [inline code/game/objects/structures/reflector.dm]: loc -> drop_location()

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by ACCount12** [comment]
All fixed. If you are trying to discourage me from ever touching the old code, you are doing great.

**Review by Cyberboss — APPROVED** [review]

**Comment by Cyberboss** [comment]
Changelog please

**Comment by ACCount12** [comment]
Please, pile all your suggestions in a single post next time. You are making this unnecessary painful for both of us.

I'll add the changelog somewhere in the next 2 hours.

**Comment by ACCount12** [comment]
Changelog is in.

**Pull request merged**